gobblin-data-management/src/main/java/gobblin/data/management/retention/profile/ConfigurableGlobDatasetFinder.java

```diff
@@ -68,7 +68,7 @@
 
   public ConfigurableGlobDatasetFinder(FileSystem fs, Properties jobProps, Config config) throws IOException {
     for (String property : requiredProperties()) {
-      Preconditions.checkArgument(config.hasPath(property) || config.hasPath(DEPRECATIONS.get(property)));
+      Preconditions.checkArgument(config.hasPath(property) || config.hasPath(DEPRECATIONS.get(property)), String.format("Missing required property %s", property));
     }
 
     if (config.hasPath(DATASET_BLACKLIST_KEY)) {
```

Comment:
While you are at it, can you also fix this -- the code seems to check only for the deprecated property but does not use  the new property DATASET_FINDER_BLACKLIST_KEY.